File: Pizzeria.Api/Pizzeria.Business/Models/Order.cs
namespace Pizzeria.Business.Models
{
    public class Order
    {
        public int Id { get; set; }

        public Restaurant? Restaurant { get; set; }

        public IEnumerable<PizzaOrder> PizzaOrders { get; set; }

        public decimal TotalCost => PizzaOrders.Sum(p => p.TotalCost);
    }
}

File: pizzeria/src/Models/Order.ts
import { PizzaOrder } from "./PizzaOrder";

export class Order {
  id: number = 0;
  restaurantId: number = 0;
  pizzaOrders: PizzaOrder[] = [];
  totalCost: number = 0;

  constructor(restaurantId: number) {
    this.restaurantId = restaurantId;
  }
}
